lib/intern.js

```diff
@@ -0,0 +1,162 @@
+const Mongo = require('mongodb')
+const ObjectID = Mongo.ObjectID
+
+
+function ensure_id(ent, opts) {
+  let id = undefined
+
+  if (undefined !== ent.id$) {
+    id = ent.id$
+  } else if (opts.generate_id) {
+    id = opts.generate_id(ent)
+  }
+
+  return makeid(id)
+}
+
+
+function makeid(hexstr) {
+  if ('string' === typeof hexstr && 24 === hexstr.length) {
+    try {
+      return ObjectID.createFromHexString(hexstr)
+    } catch (e) {
+      return hexstr
+    }
+  }
+
+  return hexstr
+}
+
+
+function idstr(obj) {
+  return obj && obj.toHexString ? obj.toHexString() : '' + obj
+}
+
+
+function fixquery(q, seneca, opts) {
+  if (q.native$) {
+    return  Array.isArray(q.native$) ? q.native$[0] : q.native$
+  }
+
+  if ('string' === typeof q) {
+    return { _id: makeid(q) }
+  }
+  
+  if (Array.isArray(q)) {
+    return {
+      _id: { $in: q.map((id) => makeid(id)) }
+    }
+  }
+
+  if (q.id) {
+    if (Array.isArray(q.id)) {
+      return {
+        _id: { $in: q.id.map((id) => makeid(id)) }
+      }
+    }
+
+    return { _id: makeid(q.id) }
+  }
+
+
+  const qq = {}
+
+  for (const qp in q) {
+    if (is_seneca_qualifier(qp)) {
```

Comment:
use "directive", not "qualifier"
"directive" is the terminology established in the Seneca docs for $ props that do special things